com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs

```diff
@@ -145,23 +171,38 @@ internal class SnapshotSystem : INetworkUpdateSystem, IDisposable
         internal SendMessageHandler SendMessage;
         internal SpawnObjectHandler SpawnObject;
         internal DespawnObjectHandler DespawnObject;
+        internal GetBehaviourVariableHandler GetBehaviourVariable;
 
         // Property showing visibility into inner workings, for testing
         internal int SpawnsBufferCount { get; private set; } = 100;
         internal int DespawnsBufferCount { get; private set; } = 100;
 
+        internal int UpdatesBufferCount { get; private set; } = 100;
+
+        internal IndexAllocator MemoryStorage = new IndexAllocator(10000, 1000);
```

Comment:
I have some concerns about rolling our own memory management primitives but I'm not entirely sure.
I had a look at `IndexAllocator` implementation and looks like we'd like to maintain a good chunk of code there.
I'll try to grab you to chat later, just a note for future self.